orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityTask.groovy

```diff
@@ -158,11 +158,17 @@ class ApplySourceServerGroupCapacityTask extends AbstractServerGroupTask {
       ancestorStage.context.containsKey("sourceServerGroupCapacitySnapshot")
     }
 
-    return ancestors ? ancestors[0] : stage.execution.stages.find {
+    Stage ancestor = (ancestors != null && !ancestors.isEmpty()) ? ancestors[0] : stage.execution.stages.find {
       // find a synthetic sibling w/ 'sourceServerGroupCapacitySnapshot' in the event of there being no suitable
```

Comment:
The return `ancestors ?` should have been doing the same as the revised not null and not empty check?